Trimble.Modus.Components/Controls/TMRadioButton/TMRadioButtonGroup.cs

```diff
@@ -0,0 +1,501 @@
+    /// </summary>
+    private static void OnSelectedIndexChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        ((TMRadioButtonGroup)bindable).UpdateSelectedStates();
+        ((TMRadioButtonGroup)bindable).SetDefaultSelectedRadioButton();
+    }
+
+    /// <summary>
+    /// Disable radio buttons in the group when the <see cref="IsEnabled"/> property changes
+    /// </summary>
+    private static void OnIsEnabledChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMRadioButtonGroup radioButtonGroup = (TMRadioButtonGroup)bindable;
+        if (radioButtonGroup._radioButtons.Count > 1)
+        {
+            foreach (TMRadioButton radioButton in radioButtonGroup._radioButtons)
+            {
+                radioButton.IsEnabled = (bool)newValue;
+            }
+        }
+        radioButtonGroup._label.Opacity = (bool)newValue ? 1 : 0.5;
+    }
+
+    /// <summary>
+    /// Update the title of the group when the <see cref="GroupTitle"/> property changes
+    /// </summary>
+    private static void OnGroupTitleChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMRadioButtonGroup radioGroup = (TMRadioButtonGroup)bindable;
+        string text = (string)newValue;
+        radioGroup._label.Text = text;
+        radioGroup._label.IsVisible = !String.IsNullOrEmpty(text);
+    }
+
+    /// <summary>
+    /// Update the size of the radio buttons when the <see cref="Size"/> property changes
+    /// </summary>
+    private static void OnSizeChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMRadioButtonGroup group = (TMRadioButtonGroup)bindable;
+        if (group._radioButtons.Count > 1)
+        {
+            foreach (TMRadioButton button in group._radioButtons)
+            {
+                button.Size = (CheckboxSize)newValue;
+            }
+        }
+        group._label.FontSize = (CheckboxSize)newValue == CheckboxSize.Default ? 14 : 16;
+    }
+
+    /// <summary>
+    /// Updates the ItemsSource
+    /// </summary>
+    private static void OnRadioItemsSourceChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMRadioButtonGroup group = (TMRadioButtonGroup)bindable;
+        if (group == null)
+        {
+            return;
+        }
+        if (oldValue is INotifyCollectionChanged oldCollection)
+        {
+            oldCollection.CollectionChanged -= group.OnItemsSourceCollectionChanged;
+        }
+        if (newValue is INotifyCollectionChanged newCollection)
+        {
+            newCollection.CollectionChanged += group.OnItemsSourceCollectionChanged;
+        }
+        group.PopulateItems();
+    }
+
+    /// <summary>
+    /// Populate radiobuttons in items source in the group
+    /// </summary>
+    private void PopulateItems()
+    {
+        if (ItemsSource == null) return;
+        foreach (var item in ItemsSource)
```

Comment:
Don't we have to remove all the old children first before adding new children?

Reply:
Yes, done